### PR TITLE
Feature: updated the documentation in the assembly index module

### PR DESCRIPTION
**Pull request by 0xSwapFeeder**

### CODE REVIEW

**Review by Elli610 — CHANGES_REQUESTED** [review]
For the "call" function : are you sure about " @param args - The serialized arguments to send to the function to call. " ?

The args parameter is also serialized while calling env.call in the return.

**Comment by 0xSwapFeeder** [comment]
You are right, didn't saw that, the old documantation was saying it was needed but this might not be the case. Fixing that ASAP.

File: assembly/std/index.ts
Comment (Elli610) [inline assembly/std/index.ts:51]:
```suggestion
 * @param args - The arguments of the function we are calling.
```

(The "args" parameter doesn't need to be serialized. It is done in the " return env.localCall(at.toString(), functionName, args.serialize());"
)

File: assembly/std/index.ts
Comment (Elli610) [inline assembly/std/index.ts:75]:
```suggestion
 * @param args - The arguments of the function we are calling.
```

(The "args" parameter doesn't need to be serialized. It is done in the "return env.localExecution(bytecode, functionName, args.serialize());")

File: assembly/std/index.ts
Comment (Elli610) [inline assembly/std/index.ts:108]:
```suggestion
 * @returns The serialized bytecode of the contract
```

File: assembly/std/index.ts
Comment (Elli610) [inline assembly/std/index.ts:26]: the argument coins refer to the tokens that we send to the contract. Their aren't the price you are paying for the execution

File: assembly/std/index.ts
Comment (Elli610) [inline assembly/std/index.ts:174]:
```suggestion
 * @remarks the caller should be approved before
 */
```

File: assembly/std/index.ts
Comment (Elli610) [inline assembly/std/index.ts:202]:
```suggestion
 * @param key - The serialized key to check for
```

File: assembly/std/index.ts
Comment (Elli610) [inline assembly/std/index.ts:217]:
```suggestion
 * @param key - The serialized key of the data to get.
```

File: assembly/std/index.ts
Comment (BatiGencho) [reply assembly/std/index.ts:458]: Ok, this function is not needed anymore in my opinion as we have a totally different approach of deploying smart contracts using our `toolkit`. It is a leftover from the times where we first had a node script to replace the bytecode in a so-called deployer contract, then we had a transformer to do that (which we still keep btw.) and finally we moved to the toolkit that uses under the hood the `sc-compiler` to do that.
I believe we could remove this one for good @Thykof @gregLibert @Ben-Rey ?

File: assembly/std/index.ts
Comment (Elli610) [inline assembly/std/index.ts:357]:
```suggestion
 * @param pubKey - Base58check encoded public key
```

**Review by Elli610 — CHANGES_REQUESTED** [review]

File: assembly/std/index.ts
Comment (Elli610) [inline assembly/std/index.ts:333]: in "return env.toBase58(data);", the toBase58 function is marked as non existant in env

File: assembly/std/index.ts
Comment (0xSwapFeeder) [reply assembly/std/index.ts]: Fixed that, it is more likely that this arguments describes coins to pass to some kind of "payable" function, can you confirm that @BatiGencho @Thykof @Ben-Rey ?

Comment (Thykof) [reply assembly/std/index.ts]: probably yes

File: assembly/std/index.ts
Comment (Ben-Rey) [reply assembly/std/index.ts:459]: Sgtm

File: assembly/std/index.ts
Comment (Ben-Rey) [inline assembly/std/index.ts:31]: Check in the storage file how we document exeptions

File: assembly/std/index.ts
Comment (Ben-Rey) [inline assembly/std/index.ts:56]: Same here

File: assembly/std/index.ts
Comment (Ben-Rey) [inline assembly/std/index.ts:78]: same here, I think it would be better to put it a @throw tag, I wouldn't put an exclamation mark at the end.

Comment (Ben-Rey) [reply assembly/std/index.ts:78]: Maybe we can check if there is more exeption like if the function doesn't exist.

File: assembly/std/index.ts
Comment (Ben-Rey) [inline assembly/std/index.ts:121]: This comment is for callerHasWriteAccess. (Idk why I can't select it).
I suggest adding more descriptive details to the function's title and/or documentation to clarify what specific resource or data the function is checking for write access. This will help improve the clarity and usefulness of the function for other developers.

File: assembly/std/index.ts
Comment (Ben-Rey) [inline assembly/std/index.ts:143]: I would change to => the address of the newly created smart contract on the ledger

File: assembly/std/index.ts
Comment (gregLibert) [reply assembly/std/index.ts:459]: Nope, we still need this in case of the enduser is not using a transformer. Let's talk about it if needed

File: assembly/std/index.ts
Comment (Thykof) [inline assembly/std/index.ts:35]:
```suggestion
 *    - the address doesn't exist!
 *    - the function doesn't exist in the contract!
```

File: assembly/std/index.ts
Comment (Thykof) [inline assembly/std/index.ts:62]:
```suggestion
 *    - the address doesn't exist!
 *    - the function doesn't exist in the contract!
```

File: assembly/std/index.ts
Comment (Thykof) [inline assembly/std/index.ts:79]:
```suggestion
 * - Runtime exception if the function doesn't exist in the bytecode!
```

File: assembly/std/index.ts
Comment (Thykof) [inline assembly/std/index.ts:116]:
```suggestion
 * Runtime exception if the address doesn't exist!
```

File: assembly/std/index.ts
Comment (Thykof) [reply assembly/std/index.ts:486]: see transformer here https://github.com/massalabs/as/tree/main/packages/as-transformer#file2bytearray

Comment (0xSwapFeeder) [reply assembly/std/index.ts]: I added the link to the as-transformer information then.

File: assembly/std/index.ts
Comment (0xSwapFeeder) [reply assembly/std/index.ts]: Done!

File: assembly/std/index.ts
Comment (0xSwapFeeder) [reply assembly/std/index.ts:181]: Tell me if this is descriptive enough now.

File: assembly/std/index.ts
Comment (Thykof) [inline assembly/std/index.ts:175]: callerHasWriteAccess will not grant you write access.

if callerHasWriteAccess returns true, then that's mean you have write access

File: assembly/std/index.ts
Comment (Thykof) [inline assembly/std/index.ts:175]: let's use https://typedoc.org/tags/see/ something like:

```
 * When calling @see{create_new_sc_address}, the User or smart contract will be granted write
```

**Comment by 0xSwapFeeder** [comment]
Setting this PR on hold, some work needs to be done first.
Re-creating one, when the work is done, with a more seem less reviewers flow later.

**Pull request closed without merging**
